pymc3/aesaraf.py

```diff
@@ -882,6 +893,19 @@ def compile_rv_inplace(inputs, outputs, mode=None, **kwargs):
     Using this function ensures that compiled functions containing random
     variables will produce new samples on each call.
     """
+
+    # If not specified, set the default_update of RandomVariables rng, so that
+    # it is automatically updated after every function call
+    output_to_list = outputs if isinstance(outputs, list) else [outputs]
+    for rv in (
+        node
+        for node in walk_model(output_to_list, walk_past_rvs=True)
+        if node.owner and isinstance(node.owner.op, RandomVariable)
+    ):
+        rng = rv.owner.inputs[0]
+        if not hasattr(rng, "default_update"):
+            rng.default_update = rv.owner.outputs[0]
```

Comment:
This is a WIP hack that seems to fix the issues I was finding in this PR, until I get to the bottom of the problem.

In the meantime and just to be sure, is it a bad idea to set the `default_updates` here before compilation. It requires a full graph transversal, but also means we could remove this code here: https://github.com/pymc-devs/pymc3/blob/c913709d02deab1307df89245782630d8f25d953/pymc3/distributions/distribution.py#L307-L322

CC @brandonwillard